dora/core/server/worker/src/main/java/alluxio/worker/netty/FileWriteHandler.java

```diff
@@ -0,0 +1,163 @@
+ */
+
+package alluxio.worker.netty;
+
+import alluxio.DefaultStorageTierAssoc;
+import alluxio.StorageTierAssoc;
+import alluxio.conf.Configuration;
+import alluxio.conf.PropertyKey;
+import alluxio.metrics.MetricsSystem;
+import alluxio.network.protocol.RPCProtoMessage;
+import alluxio.proto.dataserver.Protocol;
+import alluxio.worker.dora.DoraWorker;
+
+import com.google.common.base.Preconditions;
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.Channel;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.ExecutorService;
+import javax.annotation.concurrent.NotThreadSafe;
+
+/**
+ * This handler handles block write request. Check more information in
+ * {@link AbstractWriteHandler}.
+ */
+@alluxio.annotation.SuppressFBWarnings(
+    value = "BC_UNCONFIRMED_CAST_OF_RETURN_VALUE",
+    justification = "false positive with superclass generics, "
+        + "see more description in https://sourceforge.net/p/findbugs/bugs/1242/")
+@NotThreadSafe
+public final class FileWriteHandler extends AbstractWriteHandler<BlockWriteRequestContext> {
+  private static final Logger LOG = LoggerFactory.getLogger(FileWriteHandler.class);
+  private static final long FILE_BUFFER_SIZE = Configuration.getBytes(
+      PropertyKey.WORKER_FILE_BUFFER_SIZE);
+
+  /** The Block Worker which handles blocks stored in the Alluxio storage of the worker. */
+  private final DoraWorker mWorker;
+  /** An object storing the mapping of tier aliases to ordinals. */
+  private final StorageTierAssoc mStorageTierAssoc = new DefaultStorageTierAssoc(
+      PropertyKey.WORKER_TIERED_STORE_LEVELS,
+      PropertyKey.Template.WORKER_TIERED_STORE_LEVEL_ALIAS);
+
+  /**
+   * Creates an instance of {@link FileWriteHandler}.
+   *
+   * @param executorService the executor service to run {@link PacketWriter}s
+   * @param blockWorker the block worker
+   * @param ufsManager the UFS manager
+   */
+  FileWriteHandler(ExecutorService executorService, DoraWorker doraWorker) {
+    super(executorService);
+    mWorker = doraWorker;
+  }
+
+  @Override
+  protected boolean acceptMessage(Object object) {
+    if (!super.acceptMessage(object)) {
+      return false;
+    }
+    Protocol.WriteRequest request = ((RPCProtoMessage) object).getMessage().asWriteRequest();
+    return request.getType() == Protocol.RequestType.ALLUXIO_BLOCK;
+  }
+
+  @Override
+  protected PacketWriter createPacketWriter(BlockWriteRequestContext context, Channel channel) {
+    return new BlockPacketWriter(context, channel, mWorker);
+  }
+
+  @Override
+  protected BlockWriteRequestContext createRequestContext(Protocol.WriteRequest msg) {
+    BlockWriteRequestContext context = new BlockWriteRequestContext(msg, FILE_BUFFER_SIZE);
+    return context;
+  }
+
+  @Override
+  protected void initRequestContext(BlockWriteRequestContext context) throws Exception {
+    BlockWriteRequest request = context.getRequest();
```

Comment:
is this line needed?

Reply:
Removed. This method is reserved as a placeholder because there is no need to do any processing here